src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -1747,7 +1745,7 @@ private ValueTask<int> ReadBufferedAsync(Memory<byte> destination)
             // If the caller provided buffer, and thus the amount of data desired to be read,
             // is larger than the internal buffer, there's no point going through the internal
             // buffer, so just do an unbuffered read.
-            return destination.Length >= _readBuffer.Length ?
+            return destination.Length == 0 || destination.Length >= _readBuffer.Length ?
```

Comment:
Why does the 0 byte read case here go through the non-buffered path?

At the very least we should update the comment to explain this.

Reply:
Same as with the chunked stream, it's to ensure the zero byte read is propagated to SslStream.
BufferedRead will always request at least 4k bytes.

I will update the comment.